### PR TITLE
use new `update_called` from metrics

### PR DESCRIPTION
**Pull request by matsumotosan**
## What does this PR do?

Must be fixed for https://github.com/Lightning-AI/torchmetrics/pull/2141 to pass

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--18714.org.readthedocs.build/en/18714/

### CODE REVIEW

File: src/lightning/pytorch/trainer/connectors/logger_connector/result.py [hunk 1]
```diff
@@ -265,7 +265,7 @@ def _wrap_compute(self, compute: Any) -> Any:
         # Override to avoid syncing - we handle it ourselves.
         @wraps(compute)
         def wrapped_func(*args: Any, **kwargs: Any) -> Optional[Any]:
-            if not self._update_called:
+            if not self.update_called:
```
Comment (awaelchli) [inline src/lightning/pytorch/trainer/connectors/logger_connector/result.py:268]: The public property [was introduced in in torchmetrics 1.0.0](https://github.com/Lightning-AI/torchmetrics/commit/6e2213a65c6bbd0af975bb487759dfd009137112), but Lightning supports[ >=0.7.0](https://github.com/Lightning-AI/lightning/blob/343f80436d3233bac155e77b92891f6281ff90d1/requirements/pytorch/base.txt#L9). So there are two options I see:
A) Bump the minimum torchmetrics requirement to 1.0.0
B) Add a switch here that >= 1.0.0 will use the public attribute, while < 1.0.0 still uses the protected one.

File: src/lightning/pytorch/trainer/connectors/logger_connector/result.py
Comment (ishandutta0098) [inline src/lightning/pytorch/trainer/connectors/logger_connector/result.py:1]: I would suggest importing the packaging library here to check for the versions:
`import packaging`

File: src/lightning/pytorch/trainer/connectors/logger_connector/result.py (hunk 1 above)
Comment (ishandutta0098) [reply src/lightning/pytorch/trainer/connectors/logger_connector/result.py:268]: @matsumotosan this way you can use a switch for the torchmetrics version as suggested in B.

```suggestion
            tm_version = packaging.version.parse(torchmetrics.__version__)
            update_called_attribute = self.update_called if tm_version >= packaging.version.parse("1.0.0") else self._update_called
            if not update_called_attribute:
```

**Review by ishandutta0098 — COMMENTED** [review]
I have added the steps for using the torchmetrics version to determine whether to use the public property or the private property

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**